db/schema_tables.cc

```diff
@@ -305,23 +305,31 @@ schema_ptr tables() {
 
 // Holds Scylla-specific table metadata.
 schema_ptr scylla_tables(schema_features features) {
-    static auto make = [] (bool has_cdc_options) -> schema_ptr {
+    static auto make = [] (bool has_cdc_options, bool has_per_table_partitioners) -> schema_ptr {
```

Comment:
I'm sorry, I don't understand anything that's going on in this function, @haaawk I hope you understand what you did here :-) I seriously think at some point we'll need to create a more understandable mechanism for adding features, and document it so it doesn't look like black magic.
That being said, I am saying that I don't understand this - not that I think it's wrong. So I'll work under the assumption that Avi, who also reviewed this code, understood it and it's correct...

Reply:
I'm just extending the mechanism that was previously added for cdc_options so this change is not actually doing much and just relying on already existing mechanism.

Reply:
That's an indication that @nyh is right to worry. A change should be understood, not be such a textual extension. We should encapsulate it into a function with clear explanations.

Reply:
OK. I will as a follow-up PR

Reply:
I didn't mean that you should do it (but you're welcome to do it)